Lib/fontTools/cffLib/specializer.py

```diff
@@ -0,0 +1,499 @@
+	@staticmethod
+	def vhcurveto(args):
+		if len(args) < 4 or len(args) % 8 not in {0,1,4,5}: raise ValueError(args)
+		last_args = None
+		if len(args) % 2 == 1:
+			lastStraight = len(args) % 8 == 5
+			args, last_args = args[:-5], args[-5:]
+		it = _everyN(args, 4)
+		try:
+			while True:
+				args = next(it)
+				yield ('rrcurveto', [0, args[0], args[1], args[2], args[3], 0])
+				args = next(it)
+				yield ('rrcurveto', [args[0], 0, args[1], args[2], 0, args[3]])
+		except StopIteration:
+			pass
+		if last_args:
+			args = last_args
+			if lastStraight:
+				yield ('rrcurveto', [0, args[0], args[1], args[2], args[3], args[4]])
+			else:
+				yield ('rrcurveto', [args[0], 0, args[1], args[2], args[4], args[3]])
+
+	@staticmethod
+	def rcurveline(args):
+		if len(args) < 8 or len(args) % 6 != 2: raise ValueError(args)
+		args, last_args = args[:-2], args[-2:]
+		for args in _everyN(args, 6):
+			yield ('rrcurveto', args)
+		yield ('rlineto', last_args)
+	@staticmethod
+	def rlinecurve(args):
+		if len(args) < 8 or len(args) % 2 != 0: raise ValueError(args)
+		args, last_args = args[:-6], args[-6:]
+		for args in _everyN(args, 2):
+			yield ('rlineto', args)
+		yield ('rrcurveto', last_args)
+
+
+def generalizeCommands(commands, ignoreErrors=True):
+	result = []
+	mapping = _GeneralizerDecombinerCommandsMap
+	for op,args in commands:
+		func = getattr(mapping, op if op else '', None)
+		if not func:
+			result.append((op,args))
+			continue
+		try:
+			for command in func(args):
+				result.append(command)
+		except ValueError:
+			if ignoreErrors:
+				# Store op as data, such that consumers of commands do not have to
+				# deal with incorrect number of arguments.
+				result.append((None,args))
+				result.append((None, [op]))
+			else:
+				raise
+	return result
+
+def generalizeProgram(program, **kwargs):
+	return commandsToProgram(generalizeCommands(programToCommands(program), **kwargs))
+
+
+def _categorizeVector(v):
+	"""
+	Takes X,Y vector v and returns one of r, h, v, or 0 depending on which
+	of X and/or Y are zero, plus tuple of nonzero ones.  If both are zero,
+	it returns a single zero still.
+
+	>>> _categorizeVector((0,0))
+	('0', (0,))
+	>>> _categorizeVector((1,0))
+	('h', (1,))
+	>>> _categorizeVector((0,2))
+	('v', (2,))
+	>>> _categorizeVector((1,2))
+	('r', (1, 2))
+	"""
+	if not v[0]:
+		if not v[1]:
+			return '0', v[:1]
+		else:
+			return 'v', v[1:]
+	else:
+		if not v[1]:
+			return 'h', v[:1]
+		else:
+			return 'r', v
+
+def _mergeCategories(a, b):
+	if a == '0': return b
+	if b == '0': return a
+	if a == b: return a
+	return None
+
+def _negateCategory(a):
+	if a == 'h': return 'v'
+	if a == 'v': return 'h'
+	assert a in '0r'
+	return a
+
+def specializeCommands(commands,
+		       ignoreErrors=False,
+		       generalizeFirst=True,
+		       preserveTopology=False,
+		       maxstack=48):
+
+	maxstack -= 3 # AFDKO code uses effective 45 maxstack while the spec says 48.
```

Comment:
So is 48 a problem? Also you're decreasing the value regardless of what gets passed as parameter...